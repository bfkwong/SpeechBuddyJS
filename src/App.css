.App {
  text-align: center;
}

.TopLevelDisplay {
  padding: 15px 25px 10px 25px;
  text-align: center;
}
