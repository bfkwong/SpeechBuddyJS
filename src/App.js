import React, { useState } from "react";
import { Navbar, Nav, NavDropdown } from "react-bootstrap";
import "./App.css";
import SingleText from "./components/SingleText/SingleText";
import MultiText from "./components/MultiText/MultiText";
import TwoTxtSent from "./components/TwoTxtSent/TwoTxtSent";

const SINGLE_TEXT = "SINGLE_TEXT";
const MULTI_TEXT = "MULTI_TEXT";
const PLAGIARISM_TEXT = "PLAGIARISM_TEXT";
const SPEECH_ANALYSIS = "SPEECH_ANALYSIS";

function App() {
  const [display, setDisplay] = useState("SINGLE_TEXT");

  return (
    <div>
      <Navbar bg="light" expand="lg">
        <Navbar.Brand href="#home">Speech Buddy</Navbar.Brand>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav className="mr-auto">
            <NavDropdown title="Text Analysis" id="basic-nav-dropdown">
              <NavDropdown.Item onClick={() => setDisplay(SINGLE_TEXT)}>
                Single Text Analysis
              </NavDropdown.Item>
              <NavDropdown.Item onClick={() => setDisplay(MULTI_TEXT)}>
                Multi Text Analysis
              </NavDropdown.Item>
              <NavDropdown.Item onClick={() => setDisplay(PLAGIARISM_TEXT)}>
                Plagiarism Analysis
              </NavDropdown.Item>
            </NavDropdown>
            <NavDropdown title="Speech Analysis" id="basic-nav-dropdown">
              <NavDropdown.Item onClick={() => setDisplay(SPEECH_ANALYSIS)}>
                Speech Analysis
              </NavDropdown.Item>
              <NavDropdown.Item href="#action/3.2">
                Stopword Analysis
              </NavDropdown.Item>
            </NavDropdown>
          </Nav>
        </Navbar.Collapse>
      </Navbar>
      {display === SINGLE_TEXT && <SingleText></SingleText>}
      {display === MULTI_TEXT && <MultiText></MultiText>}
      {display === PLAGIARISM_TEXT && <TwoTxtSent></TwoTxtSent>}
    </div>
  );
}

export default App;
