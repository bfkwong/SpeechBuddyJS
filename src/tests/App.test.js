//Daniel Tisdale

import React from "react";
import { render } from "@testing-library/react";
import App from "../App";
import renderer from "react-test-renderer";
import { unmountComponentAtNode } from "react-dom";

describe("App Component", () => {
  let container;

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    if (container.hasChildNodes()) {
      unmountComponentAtNode(container);
    }
    container.remove();
    container = null;
  });

  test("App renders without failure", () => {
    render(<App />, container);
  });

  test("App snapshot test", () => {
    const appTree = renderer.create(<App />).toJSON();
    expect(appTree).toMatchSnapshot();
  });
});
