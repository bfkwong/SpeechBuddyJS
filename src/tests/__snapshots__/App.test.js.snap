// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component App snapshot test 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <a
      className="navbar-brand"
      href="/SpeechBuddyJS/home"
    >
      <img
        alt="Logo"
        src="Logo3.png"
      />
    </a>
    <a
      className="navbar-brand"
      href="/SpeechBuddyJS/home"
    >
      Speech Buddy
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <span
          className="nav-link"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <a
            href="/SpeechBuddyJS/single_text"
            onClick={[Function]}
          >
            Single Text Analysis
          </a>
        </span>
        <span
          className="nav-link"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <a
            href="/SpeechBuddyJS/two_txt_plag"
            onClick={[Function]}
          >
            Two Text Analysis
          </a>
        </span>
        <span
          className="nav-link"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <a
            href="/SpeechBuddyJS/multi_text"
            onClick={[Function]}
          >
            Multi Text Analysis
          </a>
        </span>
      </div>
    </div>
  </nav>
  <div
    className="flexContainer"
  >
    <header
      className="header"
    >
      <div
        className="main flexItem"
      >
        <h1>
          Welcome to SpeechBuddy
        </h1>
        <h5>
          A Free Text and Speech Analyzer
        </h5>
        <h6>
          For Students and Teachers Alike
        </h6>
      </div>
      <div
        className="overview flexItem"
      >
        <h2>
          Speech Buddy Overview
        </h2>
        <p>
          Speech Buddy is a text and speech analysis toolkit for quick statistics and visualizations on any given text.
        </p>
      </div>
    </header>
    <header
      className="features flexItem"
    >
      <h2>
         Analyze Text Tone & Word Variance for Single Text Speeches 
      </h2>
      <div>
        <p>
          Check the tone of your text before you perform in front of an audience. Ensure that your audience reacts the way you expect depending on your intended tone:
        </p>
        <ul>
          <li>
            <a
              href="/SpeechBuddyJS/single_text"
              onClick={[Function]}
            >
              Single Text Analysis
            </a>
          </li>
        </ul>
      </div>
      <h2>
         Plagiarism and Similarity Comparison Between 2+ Text Speeches 
      </h2>
      <div>
        <p>
          Compare two or more .txt files for speech similarity:
        </p>
        <ul>
          <li>
            <a
              href="/SpeechBuddyJS/two_txt_plag"
              onClick={[Function]}
            >
              Two Text Analysis
            </a>
          </li>
          <li>
            <a
              href="/SpeechBuddyJS/multi_text"
              onClick={[Function]}
            >
              Multi Text Analysis
            </a>
          </li>
        </ul>
      </div>
    </header>
    <header
      className="reviews flexItem"
    >
      <h2>
        Satisfied User Reviews
      </h2>
      <div>
        <img
          alt="5Stars"
          id="stars"
          src="Stars.png"
        />
        <p>
          "SpeechBuddy is wonderful! I used to have to manually check each essay for plagiarism but now that it's automated I don't have to! I give it a 9/9 for quality tools and a 1/1 for aesthetic. Total 10/10!"
        </p>
        <span>
          - Karen J. (High School Teacher)
        </span>
        <img
          alt="5Stars"
          id="stars"
          src="Stars.png"
        />
        <p>
          "This was a very helpful tool when writing speeches for my Speech Writing class. The sentiment and toxicity analysis helps me better gauge how my audience would feel during my presentation! Definitely recommend utilizing before a big presentation!"
        </p>
        <span>
          - Smar T. (Harvard Student)
        </span>
        <img
          alt="5Stars"
          id="stars"
          src="Stars.png"
        />
        <p>
          "My dad like this alot."
        </p>
        <span>
          - Smar T. Jr. (Preschool Student)
        </span>
      </div>
    </header>
  </div>
</div>
`;
